docs/API.md

```diff
@@ -28,15 +25,24 @@ class Dirty
 end
 END
 
-source_code = Reek::Source::SourceCode.from(source)
 reporter = Reek::CLI::Report::TextReport.new
-reporter.add_examiner Reek::Core::Examiner.new(source_code)
+examiner = Reek::Examiner.new(source)
+reporter.add_examiner examiner
+
 puts reporter.show
 ```
 
 This will show the list of errors in variable `source`.
 
-`reek` can take `source` as `String`, `File` or `IO`.
+`Reek::Examiner.new` can take `source` as `String`, `File` or `IO`, or an array
+of file names.
+
+```
+# Examine a file object
+reporter.add_examiner Reek::Examiner.new(File.new('dirty.rb'))
+# Examine source from an array of file names
+reporter.add_examiner Reek::Examiner.new(['also_dirty.rb', 'clean.rb')
+```
```

Comment:
Hmmmmm. Did you take @chastell / my suggestion from here: https://github.com/troessner/reek/issues/526

```Ruby
report = Reek::API.report(source:   source,
                          reporter: :text,
                          smells:   %i(attribute data_clump nil_check),
                          config:   config) # a path to a YAML file or a Hash
puts report.show
```

into account as well? Or rather: Are you going to?
I really think we should get rid of this "report.add_..." thingie for our public API.
Or should I not even comment on that since it's WIP?..:)